auto_gpt_benchmarking/AutoGPTAgent.py

```diff
@@ -0,0 +1,91 @@
+"""
+This instantiates an AutoGPT agent who is capable of handling any task.
+It is designed to pass benchmarks as effectively as possible.
+
+Loads in the ai_settings.yaml file to get the AI's name, role, and goals.
+Sets the ai to continuous mode, but kills it if it takes more than 50,000 tokens on any particular evaluation.
+
+The model is instantiated with a prompt from the AutoGPT completion function.
+
+Eventualy we will also save and log all of the associated output and thinking for the model as well
+"""
+from pathlib import Path
+import os
+
+
+class AutoGPTAgent:
+    """
+    A class object that contains the configuration information for the AI
+    The init function takes an evaluation prompt.
+    It copies the ai_settings.yaml file in AutoGPTData to the Auto-GPT repo.
+    It then copies the given prompt to a text file to Auto-GPT/auto_gpt_workspace called prompt.txt
+    It then polls the token usage of the model and for a file called output.txt in the Auto-GPT/auto_gpt_workspace folder.
+    If the model has used more than 50,000 tokens, it kills the model.
+    If the model has used less than 50,000 tokens, it returns the output.txt file.
+    """
+    def _clean_up_workspace(self):
```

Comment:
Need to change this to clear everything out of the dir. This is brittle and other files sometimes get created confusing future agents.